Makefile

```diff
@@ -19,21 +22,51 @@ $(error COMMIT_HASH was not properly set)
 endif
 
 
+## Functions
+get_github_latest_version = $$(docker run --rm easy_infra:latest "curl -s https://api.github.com/repos/$(1)/releases/latest | jq -r '.tag_name'")
+update_dockerfile_package = ./update.sh --package=$(1) --version=$(2)
+update_dockerfile_repo    = ./update.sh --repo=$(1) --version=$(2)
+
+
 ## Rules
+.PHONY: all
+all: update build
+
+.PHONY: update
+update: update-functions update-deps
+
+.PHONY: update-deps
+update-deps: update-apt update-awscli update-github update-terraform update-yarn
+
+
+.PHONY: update-functions
+update-functions:
+	@echo "Updating the functions script..."
+	@echo "# DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN" > functions
+	@cat functions_static >> functions
+	@chmod 0755 functions
+	@# This method works as long as FUNCTION_GENERATOR is treated like a list of 4-tuples
+	@# It was chosen because Make and bash don't natively support multidimensional arrays
+	@i=0; for element in $(FUNCTION_GENERATOR); do \
+		if (( i % 4 == 0 )) ; then \
```

Comment:
I'm having trouble understanding this check. Why isn't the warning line just printed once before the `for` loop?

Reply:
I'm leaning towards over-communicating.  This approach makes sure the notice gets added before the static template and then directly before each generated function.